backend/nokia-innovative-project/src/test/java/com/nokia/library/nokiainnovativeproject/controllers/BookControllerTest.java

```diff
@@ -0,0 +1,92 @@
+package com.nokia.library.nokiainnovativeproject.controllers;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.nokia.library.nokiainnovativeproject.utils.Mappings;
+import lombok.RequiredArgsConstructor;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.DEFINED_PORT)
+@RequiredArgsConstructor
```

Comment:
I am sure that all tests fail.
Still you use default configuration and in default configuration you do not have any database configuration.